### ISSUE TITLE
Create simple and simpler variants for SwitchField

### ISSUE THREAD

**Issue by dhruvkb** [creativecommons/cc-vocabulary#25]
Most components have a default version and two simplified versions, namely `slight`ly simple and `extreme`ly simple. The SwitchField is one that does not have these variants. The goal is to develop them.

**Comment by dhruvkb** [comment]
To be fair, a lot of components have little stylistic diversity. Expanding the issue to list and discuss these components.

The following is a component and stylistic variant matrix. The goal is for all components to have all variants checkmarked. Please add more components that lack variants.

| Component | Simplicity | Roundness |
| - | - | - |
| Button | <li>- [x] Done</li> | <li>- [x] Done</li> |
| ChoiceField | <li>- [x] Done</li> | |
| InputField | <li>- [x] Done</li> | <li>- [x] Done</li> |
| ProgressBar | <li>- [ ] Done</li> | <li>- [x] Done</li> |
| SelectField | <li>- [x] Done</li> | <li>- [x] Done</li> |
| Shield | <li>- [x] Done</li> | <li>- [x] Done</li> |
| Statistic | <li>- [ ] Done</li> | <li>- [ ] Done</li> |
| SwitchField | <li>- [x] Done</li> | <li>- [x] Done</li> |

**Comment by loranazarenko** [comment]
Hi, I'm an Outreachy applicant. I would like to work on this issue, if possible. I don’t know Vue, but I studied a bit of javascript.

**Comment by dhruvkb** [comment]
Sure, go ahead! Feel free to communicate on Slack if you need any help getting started with Vue.

**Comment by loranazarenko** [comment]
I think these fields are:
Shield
Paragraph
LicenseIconography
ChoiceField
Heading
have a small selection of styles. I think soon i will load PR.

**Comment by dhruvkb** [comment]
It would be nice if you could send in some mockups or screenshots of your new styles before submitting the code.

**Comment by loranazarenko** [comment]
State 1:
![image](https://user-images.githubusercontent.com/5892607/66598806-86337e00-ebaa-11e9-94a8-55015d244e90.png)

State 2:
![image](https://user-images.githubusercontent.com/5892607/66598894-baa73a00-ebaa-11e9-9039-979b1532f804.png)

**Comment by dhruvkb** [comment]
Why are the switches green without having set the prop `color="green"`? Just to clarify, the prop `simplicity` deals with the visual prominence of a component.
![image](https://user-images.githubusercontent.com/16580576/66620495-4ac29f00-ebfe-11e9-963e-95f073da04a3.png)

**Comment by dhruvkb** [comment]
> I think these fields are:
> Shield
> Paragraph
> LicenseIconography
> ChoiceField
> Heading
> have a small selection of styles. I think soon i will load PR.

I disagree with some of your selections. `Paragraph`, `LicenseIconography` and `Heading` cannot be rounded or simplified. However I updated the issue to list the components that can be styled and yet aren't.

**Comment by loranazarenko** [comment]
> Why are the switches green without having set the prop `color="green"`? Just to clarify, the prop `simplicity` deals with the visual prominence of a component.

Because I wrote the styles in the file SwitchField.styl: 
 &.slightly-simple
    --switch-field-off-background-color $color-context-positive-light
    --switch-field-on-background-color $color-context-light
    --switch-field-handle-color $color-context-probably-light
     
  &.extremely-simple
    --switch-field-off-background-color $color-context-positive
    --switch-field-on-background-color $color-context-positive-light
    --switch-field-handle-color $color-context-probably

Do you approve of this style?

**Comment by loranazarenko** [comment]
What to do with the Shield field?  There are no simplicity styles.
ImageView - I think we can use the Roundness style.
And maybe ProgressBar too - add some transparency?

**Comment by dhruvkb** [comment]
@loranazarenko I fear I might not be making my point clearly so here are some designs so as to point you in my direction.

As we move right, the roundness goes from slight to complete. As we move down the simplicity goes from slight to extreme.
![image](https://user-images.githubusercontent.com/16580576/66694698-16b9ad80-ecd4-11e9-82da-ad2427a2d350.png)

As you can see, using a different stylistic variant should not affect the color choices.

**Comment by dhruvkb** [comment]
@loranazarenko the ImageView already has a rounded variant which might have slipped by you. Transparency doesn't really align with the general design scheme (I'm not much of a designer and a poll on Slack #usability would yield more professional thoughts) although the progress bar could definitely use some simpler variations.

Here are some ideas for Shield in terms of simplicity.
![image](https://user-images.githubusercontent.com/16580576/66694924-1ec71c80-ecd7-11e9-860f-717c9053e1c6.png)

**Comment by loranazarenko** [comment]
I was very stupid, but corrected this error and tried to use your design, that's what I got. 

![image](https://user-images.githubusercontent.com/5892607/66707673-d9ddcd00-ed4c-11e9-9503-ebd9b1000c90.png)

**Comment by loranazarenko** [comment]
I will try to do Shield.

**Comment by loranazarenko** [comment]
What about this variant:

![image](https://user-images.githubusercontent.com/5892607/66778532-9ada8380-eed4-11e9-967b-88daa64eedf1.png)

**Comment by dhruvkb** [comment]
👍 They’re much better than what I could have come up with!

Make sure you flip the greys in the inverted variant.

**Comment by loranazarenko** [comment]
I specifically left gray, maybe. Does he look bad? If you do not like it, write in which item I will correct it.

**Comment by dhruvkb** [comment]
@loranazarenko I think you've misunderstood. I'm not criticising the design, just saying that light grey on white background should automatically switch to dark grey on black background.

**Comment by loranazarenko** [comment]
Yes, all colors are invertible because they are defined through variables.

**Comment by dhruvkb** [comment]
That was my main concern. Make sure you make separate PRs per component. Would make it easier to review.

**Comment by dhruvkb** [comment]
Updated https://github.com/creativecommons/cc-vocabulary/issues/25#issuecomment-528751591 to reflect merged status of PR creativecommons/vue-vocabulary#52 and creativecommons/vue-vocabulary#53.

**Comment by loranazarenko** [comment]
Hello! Can I get on your project as an intern?

**Comment by dhruvkb** [comment]
@loranazarenko that will depend on a number of factors such as your proposal, your application and the number of seats allocated to the project. More on that can be found [here](https://opensource.creativecommons.org/programs/outreachy/2019-12-start/).

**Comment by loranazarenko** [comment]
I'm sorry, can I ask for advice  what I can do on an internship and what improvements or new features I can do.

**Comment by dhruvkb** [comment]
@loranazarenko please join Slack so that we may discuss the specifics of the Outreachy internship over chat. Instructions for joining are available [here](https://opensource.creativecommons.org/community/).
